userland/libnrfserialization/Makefile.app

```diff
@@ -18,6 +18,9 @@ endif
 
 HDRS += $(LIBNRFSER_DIR)/headers
 
+$(C_SRCS):	$(HDRS)
```

Comment:
Yeah, this makes way more sense to me. If you comment it to say something like "The C files include these headers, so we list them as an explicit dependency to ensure they are created" I will approve this PR and go away : )